### PR TITLE
SITES-732, SITES-843, SITES-858

### PR DESCRIPTION
**Pull request by jbickar**
# READY FOR REVIEW

# Summary
- Update acsf.aliases.drushrc.php to reflect `*.sites.stanford.edu` namespacing
- Update migration scripts so that people sites get sunetid.people.stanford.edu as domain, and Stack 3
- Replace all instances of cardinalsites.stanford.edu with sites.stanford.edu in migration scripts

# Needed By (Date)
- Soooon

# Criticality
- How critical is this PR on a 1-10 scale? 10/10
- The old scripts will be pretty broken once I refresh `test` and `dev`, because new sites no longer live in the cardinalsites.stanford.edu namespace.

# Steps to Test

1. Check out this branch
2. Migrate sites using the inventory that I am including at the bottom of this PR description
3. Ensure that everything migrates cleanly
4. Check sites for broken image links
5. Manually review changes to scripts for clarity and comment accuracy.

# Affected Projects or Products
- ACSF

# Associated Issues and/or People
- JIRA tickets
    -https://stanfordits.atlassian.net/browse/SITES-732
    -https://stanfordits.atlassian.net/browse/SITES-843
    -https://stanfordits.atlassian.net/browse/SITES-858

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

# Inventory
```
[people]
jbickar

[hsdo]
jumpstart-hcbss vhost="buddhiststudies"

[depts]
encumbrances-intranet

[groups]
sws-migration-testing vhost="sws-oddball-vhost"

[people:vars]
group_ids=656
ignore_updb_errors="TRUE"
php_candidates_consequence="continue"

[hsdo:vars]
group_ids=606
ignore_updb_errors="TRUE"
php_candidates_consequence="continue"

[products:children]
hsdo

[depts:vars]
group_ids=206
ignore_updb_errors="TRUE"
php_candidates_consequence="continue"

[groups:vars]
group_ids=206
ignore_updb_errors="TRUE"
php_candidates_consequence="continue"
```

### CODE REVIEW

**Comment by sherakama** [comment]
I am not able to get anything at the `*-dev.people.stanford.edu` to resolve nor does my drush alias work when I set the uri to use `people`. I have to do everything through sites right now.

**Comment by jbickar** [comment]
This may have tighter coupling to SITES-857 than I thought.

**Comment by sherakama** [comment]
No, I don't think that was it. It seems that it may have been that swsmigrationtesting and the others were missing their `sites.stanford.edu` custom domain. I switched my drush aliases over to use that domain as the uri. 

I am running the full-migration playbook and I guess because the sites already exist they did not get the custom domain.

File: roles/define-paths/tasks/main.yml [hunk 1]
```diff
@@ -43,17 +42,12 @@
 # Ex. /var/www/html/cardinald7.02dev/scripts
 - name: Set scripts_dir variable
   set_fact:
-    scripts_dir: "/var/www/html/{{ stack }}.02{% if acsf_environment=='' %}live{% elif acsf_environment=='test'%}test{% elif acsf_environment=='dev'%}dev{% endif %}/scripts"
+    scripts_dir: "/var/www/html/{{ stack }}.0{{ stack_id }}{% if acsf_environment=='' %}live{% elif acsf_environment=='test'%}test{% elif acsf_environment=='dev'%}dev{% endif %}/scripts"
```
Comment (sherakama) [inline roles/define-paths/tasks/main.yml]: Why don't we define the whole stack id. It seems silly to assume 0 here.

File: roles/setup-local/templates/acsf.aliases.drushrc.php [hunk 2]
```diff
@@ -63,8 +63,15 @@ function ($var) use ($alias_key) {
 $stack_id = array(
   "cardinalsites" => "01",
   "cardinald7" => "02",
+  "leland" => "03",
 );
 
+// Set the string for sites.stanford.edu or people.stanford.edu.
+$sites_people = "sites";
```
Comment (sherakama) [inline roles/setup-local/templates/acsf.aliases.drushrc.php]: Oh, I took a slightly different approach. 

For flexibility I would assume less about the urls. Check out https://github.com/SU-SWS/acsf-cardinald7/pull/80

File: roles/setup-site/tasks/main.yml [hunk 3]
```diff
@@ -77,7 +78,7 @@
   retries: "{{ wait_time }}"
   delay: 50
   register: bootstrap_status
-  until: bootstrap_status.stdout | search('Successful')
+  until: bootstrap_status.stdout is search('Successful')
```
Comment (sherakama) [inline roles/setup-site/tasks/main.yml]: Is this a ansible version change thing? Why the change in the pipe?

File: roles/setup-site/tasks/main.yml [hunk 4]
```diff
@@ -48,9 +49,9 @@
     site_exists: FALSE
   when: search_for_site|length == 0
 
-# - name: debug
-#   debug:
-#     var: site_exists
+- name: debug
```
Comment (sherakama) [inline roles/setup-site/tasks/main.yml]: I'd like to change our debug statements to only print when a variable is set to true.

File: roles/get-public-files-path/tasks/main.yml [hunk 5]
```diff
@@ -30,4 +30,4 @@
 - name: Be sure response does not include drush warnings
   fail:
     msg: "There appear to be drush warning messages cluttering the file_public_path variable"
-  when: file_public_path.stdout | search('The following module is missing')
+  when: file_public_path.stdout is search('The following module is missing')
```
Comment (sherakama) [inline roles/get-public-files-path/tasks/main.yml]: Why the change here?

File: roles/add-vhost/tasks/main.yml [hunk 6]
```diff
@@ -50,7 +52,7 @@
     force_basic_auth: yes
     body_format: json
     body:
-      domain_name: "{{ inventory_hostname }}{{ stanford_environment }}.cardinalsites.stanford.edu"
+      domain_name: "{{ inventory_hostname }}{{ stanford_environment }}.{{ sites_service }}.stanford.edu"
```
Comment (sherakama) [inline roles/add-vhost/tasks/main.yml]: We may as well define the full url path in our configuration instead of assuming `.stanford.edu` as we have already gone as far as to create a variable for it.

File: roles/setup-site/tasks/main.yml (hunk 3 above)
Comment (jbickar) [reply roles/setup-site/tasks/main.yml]: I was getting deprecated warnings

File: roles/define-paths/tasks/main.yml (hunk 1 above)
Comment (jbickar) [reply roles/define-paths/tasks/main.yml]: Because we use `stack_id` [here](https://github.com/SU-SWS/ansible-playbooks/blob/SITES-843-people-leland-stack/roles/setup-site/tasks/main.yml#L65) also.

File: roles/setup-local/templates/acsf.aliases.drushrc.php (hunk 2 above)
Comment (jbickar) [reply roles/setup-local/templates/acsf.aliases.drushrc.php]: Yeah, I think I'm going to poach what you did in https://github.com/SU-SWS/acsf-cardinald7/pull/80. Don't let the aliases file hold up this PR; I can do a separate one to catch up the aliases.

File: roles/setup-site/tasks/main.yml (hunk 4 above)
Comment (jbickar) [reply roles/setup-site/tasks/main.yml]: I didn't mean to leave this uncommented. Bad developer; no cookie.

File: roles/add-vhost/tasks/main.yml (hunk 6 above)
Comment (jbickar) [reply roles/add-vhost/tasks/main.yml]: I think we're using `sites_service` elsewhere. Although I suppose we could append `.stanford.edu` in the definition of the variable.

File: roles/define-paths/tasks/main.yml (hunk 1 above)
Comment (sherakama) [reply roles/define-paths/tasks/main.yml]: I guess if we get over 10 stacks and still need these scripts we are in trouble anyway.

File: roles/get-sitename/tasks/main.yml [hunk 7]
```diff
@@ -142,10 +170,12 @@
...
-# - name: Debug site_id
-#   debug:
-#     var: site_id
-#
-# - name: Acsf site name
-#   debug:
-#     var: acsf_site_name
+- name: Debug site_id
+  debug:
+  var: site_id
```
Comment (sherakama) [inline roles/get-sitename/tasks/main.yml]: This needs to be indented.

File: roles/setup-site/tasks/main.yml (hunk 3 above)
Comment (sherakama) [reply roles/setup-site/tasks/main.yml]: Good to know. Thank you.

**Comment by sherakama** [comment]
Ok, this looks good to go but I am going to hold off merging until I test and merge the code to get the *.people.stanford.edu domain in place on the factory hook.

File: roles/get-sitename/tasks/main.yml (hunk 7 above)
Comment (jbickar) [reply roles/get-sitename/tasks/main.yml]: I'm not sure how this made it in here like this. I blame Russian hackers.

Comment (sherakama) [reply roles/get-sitename/tasks/main.yml]: It is a good thing I caught it. This is how it all starts. Next thing you know, Kanye is president.

**Comment by jbickar** [comment]
`SITES-857` is blocked. Let's not let this sit too long.

**Comment by jbickar** [comment]
Oh, I have some WIP to add a custom domain for `people` sites. Let me push that up as part of this. There may be some order-of-operations issues related to the drush aliases, but it should help ease some of the pain.

**Comment by jbickar** [comment]
w00t. Note that I snuck https://github.com/SU-SWS/ansible-playbooks/pull/53/commits/f955ea925a1a1422d9d52b90b144cd6d8e2c5753 into this.

I'll update the base branch for #55
